### PR TITLE
fix: mempool lane size check on `CheckTx`

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-1473

### CODE REVIEW

File: abci/checktx/mempool_parity_check_tx.go [hunk 1]
```diff
@@ -98,10 +106,99 @@ func (m MempoolParityCheckTx) CheckTx() CheckTx {
...
+	}
+
+	// Set the remaining important context values.
+	ctx = ctx.
+		WithTxBytes(req.Tx).
+		WithEventManager(sdk.NewEventManager()).
+		WithConsensusParams(m.baseApp.GetConsensusParams(ctx))
+
+	return ctx
+}
```
Comment (davidterpay) [inline abci/checktx/mempool_parity_check_tx.go:204]: can you benchmarket how long this takes just so we have a reference on the slowdown we expected? I imagine some of this is not necessary.

File: abci/checktx/mempool_parity_check_tx.go [hunk 2]
```diff
@@ -98,10 +106,99 @@ func (m MempoolParityCheckTx) CheckTx() CheckTx {
...
+			)
+
+			return sdkerrors.ResponseCheckTxWithEvents(
+				fmt.Errorf("tx size exceeds max block bytes"),
+				0,
+				0,
+				nil,
+				false,
+			), nil
+		}
```
Comment (davidterpay) [inline abci/checktx/mempool_parity_check_tx.go:141]: Does it make sense to do this check before the entire ante-handler sequence is run?

**Review by davidterpay — APPROVED** [review]

File: abci/checktx/mempool_parity_check_tx.go [hunk 3]
```diff
@@ -98,10 +106,99 @@ func (m MempoolParityCheckTx) CheckTx() CheckTx {
 			}
 		}
 
+		sdkCtx := m.GetContextForTx(req)
```
Comment (nivasan1) [inline abci/checktx/mempool_parity_check_tx.go:109]: Why do we need to do this?

Comment (aljo242) [reply abci/checktx/mempool_parity_check_tx.go:109]: we need a context to get consensus params

File: abci/checktx/mempool_parity_check_tx.go [hunk 4]
```diff
@@ -98,10 +106,99 @@ func (m MempoolParityCheckTx) CheckTx() CheckTx {
...
+			return sdkerrors.ResponseCheckTxWithEvents(
+				err,
+				0,
+				0,
+				nil,
+				false,
+			), nil
+		}
+
+		consensusParams := sdkCtx.ConsensusParams()
```
[the remark this answers is not in the document]
Comment (aljo242) [reply abci/checktx/mempool_parity_check_tx.go:122]: I'm not sure what you're asking here.

We're getting the context off of the baseapp everytime this is run, so we have a fresh view of the consensus params. Regardless of if it is Check or Recheck

File: abci/checktx/mempool_parity_check_tx.go (hunk 2 above)
Comment (aljo242) [reply abci/checktx/mempool_parity_check_tx.go]: I'm not sure - I think the case with a huge Tx is much less likely than the case where youre Tx just fails the antehandler.  So i would rather run that and have the matching logic done later only when we need to

File: abci/checktx/mempool_parity_check_tx.go (hunk 1 above)
Comment (aljo242) [reply abci/checktx/mempool_parity_check_tx.go]: How would you imagine benchmarking this?

Start a small in mem test app and call this on it?

File: abci/checktx/mempool_parity_check_tx.go (hunk 3 above)
Comment (aljo242) [reply abci/checktx/mempool_parity_check_tx.go:109]: and to call `Match` on the lanes

File: abci/checktx/mempool_parity_check_tx.go (hunk 1 above)
Comment (davidterpay) [reply abci/checktx/mempool_parity_check_tx.go]: benchmarking with a real simapp would be better, but probably not important to get this shipped. fine as is

File: abci/checktx/mempool_parity_check_tx.go (hunk 3 above)
Comment (nivasan1) [reply abci/checktx/mempool_parity_check_tx.go:109]: I see we're wrapping the `CheckTx` handler and don't have access to the base-app state directly..

File: abci/checktx/mempool_parity_check_tx.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (nivasan1) [reply abci/checktx/mempool_parity_check_tx.go:122]: I'm asking what happens in the case that  the tx is inserted into the app-side mempool initially. But later on, the consensusParams are changed, and the tx is re-checked.

Where are we enforcing the removal of this transaction from the app-side mempool on re-check

**Review by nivasan1 — APPROVED** [review]
lgtm

**Pull request merged**